silk/tests/openthread/ot_test_multicast_traffic.py

```diff
@@ -0,0 +1,265 @@
+            msg_len=30,
+            mcast_hops=5,):
+        """
+        Send a multicast message with given `len` from `src_node` using `src_addr` to the multicast address `mcast_addr`.
+        Verify that the message is received on all nodes in `recving_nodes` list and that it is not received on all
+        nodes in `non_recving_nodes` list.
+        """
+        timeout = 10
+        delay = 1
+
+        port = random.randint(10000, 10099)
+        message = random_string(msg_len)
+        if src_addr == src_node.ip6_lla:
+            src_addr = f"{src_addr}%{src_node.netns}"
+
+        for receiver in recving_nodes:
+            receiver.receive_udp_data(port, message, timeout)
+
+        for listener in non_recving_nodes:
+            listener.receive_no_udp_data(port, message, timeout)
+
+        time.sleep(delay)
+
+        src_node.send_udp_data(target=mcast_addr, port=port, message=message, source=src_addr, hop_limit=mcast_hops)
+        time.sleep(timeout - delay)
+
+    @testcase.test_method_decorator
+    def test01_pairing(self):
+        for node1, node2 in [(self.r1, self.r2), (self.r2, self.r3), (self.r3, self.r4),
+                             (self.r2, self.fed), (self.r4, self.sed)]:
+            node1.allowlist_node(node2)
+            node2.allowlist_node(node1)
+
+        self.r1.form(self.network_data, "router")
+        self.r1.permit_join(3600)
+        self.wait_for_completion(self.device_list)
+
+        self.logger.info(self.r1.ip6_lla)
+        self.logger.info(self.r1.ip6_thread_ula)
+
+        self.network_data.xpanid = self.r1.xpanid
+        self.network_data.panid = self.r1.panid
+
+        for node in [self.r2, self.r3, self.r4]:
+            node.join(self.network_data, "router")
+            self.wait_for_completion(self.device_list)
+
+        self.fed.join(self.network_data, "end-node")
+        self.wait_for_completion(self.device_list)
+
+        self.sed.join(self.network_data, "sleepy-end-device")
+        self.sed.set_sleep_poll_interval(POLL_INTERVAL)
+        self.wait_for_completion(self.device_list)
+
+        for node in [self.r2, self.r3, self.r4]:
+            self.assertTrue(node.get(wpan.WPAN_NODE_TYPE) == wpan.NODE_TYPE_ROUTER)
+
+        r2_ext_address = self.r2.getprop(wpan.WPAN_EXT_ADDRESS)[1:-1]
+        fed_parent = self.fed.getprop(wpan.WPAN_THREAD_PARENT)[1:17]
+        self.assertTrue(fed_parent == r2_ext_address)
+
+        r4_ext_address = self.r4.getprop(wpan.WPAN_EXT_ADDRESS)[1:-1]
+        sed_parent = self.sed.getprop(wpan.WPAN_THREAD_PARENT)[1:17]
+        self.assertTrue(sed_parent == r4_ext_address)
+
+    @testcase.test_method_decorator
+    def test02_verify_multicast_all_nodes(self):
+        #
+        #    r1 ---- r2 ---- r3 ---- r4
+        #            |               |
+        #            |               |
+        #           fed             sed
+        #
+        for node1, node2 in [(self.r1, self.r3), (self.r1, self.fed), (self.r3, self.fed),
```

Comment:
Added extra allowlist for the nodes as the test was failing saying unknownneighbor.
i.e.
[2020-10-26 23:10:13,619] [silk.Dev-5B97.wpantund] [DEBUG] wpantund[22295]: NCP => [0000000457737][INFO]-MAC-----: Frame rx failed, error:UnknownNeighbor

Adding this allowlisting resolved error:UnknownNeighbor.

Reply:
Should not need to add extra allowlist. Can you please check the route table ?

Reply:
Ack